text/src/autogluon/text/text_prediction/mx/models.py

```diff
@@ -777,6 +802,34 @@ def get_recommended_resource(nthreads_per_trial=None,
     return nthreads_per_trial, ngpus_per_trial
 
 
+def update_legacy_cfg(cfg, version_id):
+    """
+
+    Parameters
+    ----------
+    cfg
+        The configuration
+    version_id
+        The version ID
+
+    Returns
+    -------
+    new_cfg
+        The fixed configuration
+    """
+    if py_version.parse(version_id) >= py_version.parse('0.4.0'):
+        return cfg
+    else:
+        new_cfg = cfg.clone()
+        new_cfg.defrost()
+        if len(cfg.optimization.optimizer_params) > 0:
+            if isinstance(cfg.optimization.optimizer_params[0], str):
+                fixed_optimizer_params = ast.literal_eval('[' + '. '.join(cfg.optimization.optimizer_params) + ']')
+                new_cfg.optimization.optimizer_params = fixed_optimizer_params
+        new_cfg.freeze()
+        return new_cfg
+
+
 @use_np
 class MultiModalTextModel:
     """Learner of the multimodal text data.
```

Comment:
documentation for this model could be more specific, eg:  "This class implements a single neural network model that can operate on data with multiple text columns as well as tabular numeric/categorical columns."

Reply:
Added.